### ISSUE TITLE
Flutter web release builds missing data

### ISSUE THREAD

**Issue by rich-j** [Dana-Ferguson/time_machine#25]
Time Machine works great for Flutter iOS/Android and even AngularDart (one web platform). I'm sure you're aware that there's a new web platform that is looking quite promising - Flutter web. Time Machine works great with Flutter web debug builds that use DDC and serve files from packages.

However, our Flutter web release build threw the cryptic exception `Instance of 'minified:me'`. We traced this exception back to Time Machine failing to retrieve the `tzdb.json` data file. Underlying this issue is that Flutter web release builds don't have and/or correctly fetch the data files. This isn't particularly a Time Machine failing since Flutter web is a new platform and the "rules" are still being defined. 

We found that we could copy the Time Machine data files to the expected location. The expected location for `tzdb.json` is `http://myserver.com/packages/time_machine/data/tzdb/tzdb.json` with the other time zone and culture files under the `data` directory too. We also configured our CI web build to copy these data files.

So - what's the issue? We appreciate that Time Machine for the web doesn't just download the 1MB of Flutter asset `.bin` files and instead only fetches the needed zone and culture data. We also found that the Flutter team is still trying to resolve issues regarding package assets (e.g. https://github.com/flutter/flutter/issues/27802). Maybe the above "workaround" is good enough and just needs to be documented.

**Comment by GabrielBottari** [comment]
@rich-j How did you configure your CI to make it work? I had it copy the file to `build/web/assets/packages/time_machine/data/` for example. Which cloud provider are you using? Is it just a matter of serving it statically to `http://myserver.com/packages/time_machine/data/tzdb/tzdb.json` ?

**Comment by rich-j** [comment]
@GabrielBottari  Yes, we just statically serve the entire Time Machine data directory under that path. The Chrome browser developer tools network panel will let you see the paths that are trying to be accessed. Just make sure the files are available on that path.

We are still in development so we don't have a live site to point you to. For CI we use GitHub actions and deploy to Google App Engine. Our build checkouts Time Machine into a subdirectory and then copies  (along with appropriate mkdir) the data directory into our build directory `app/build/web`:

      - name: Copy time_machine data to the build area
        run: mkdir -p app/build/web/packages/time_machine/data && cp -R time_machine/lib/data app/build/web/packages/time_machine

Upon deployment the `packages` directory is top level.

**Comment by kevan-mtl** [comment]
hi @rich-j ,
Sorry to bother you again, I just want to make sure I understand the workaround properly (noob alert haha):

I copied the content of time_machine/lib/data (i.e cultures, tzdb folders + zones.json) to myapp/build/web/packages/time_machine/data/

E.g:
<img width="391" alt="Screen Shot 2020-06-07 at 10 33 30 PM" src="https://user-images.githubusercontent.com/25143586/83988027-07950900-a910-11ea-929b-490adbfcad08.png">

However, if I try to run flutter build web, since the builder utilizes the myapp/build/web to prepare all the required files for deployment, it seems to wipe the time_machine files and we are back to square 1.

Assuming I understand the first step correctly, how do you implement the rest of your workaround? (I'm using Firebase hosting in case it might be relevant)

**Comment by rich-j** [comment]
@kevan-mtl it's difficult to remotely diagnose another project's build issues. Also the Flutter build processing (particularly web) has changed over the last several months. Once our build is working satisfactorily we move on - until it breaks again and then we fix it and move on again...

For Flutter web there are 2 different compilers: Dart Development Compiler (`DDC`) and `dart2js` (for release builds). When developing (we use IntelliJ) the tools supporting `DDC` serve Flutter `assets` in a virtual `packages` directory. So if you declare the Time Machine assets for iOS/Android you should be fine running under `DDC` - in `pubspec.yaml`:

```yaml
flutter:
  assets:
    - assets/
    - packages/time_machine/data/cultures/cultures.bin
    - packages/time_machine/data/tzdb/tzdb.bin
```

BTW - the above two iOS/Android `.bin` asset files are almost 1MB combined. It's good those are not used for the web.

For web builds with `--release` Flutter uses the `dart2js` compiler and doesn't serve the virtual directories. You need to provide the data files where they're expected (see other comments above). Our CI build process copies the data files *after* the Flutter build.

One other way to get files into your web directory on build is to put those files into the Flutter top-level `web` directory alongside your `index.html`. Flutter copies all files in the `web` directory during build (a relatively recent Flutter build enhancement). The downside to this is when the Time Machine data files are updated you will need to re-copy them.

**Comment by kevan-mtl** [comment]
hi @rich-j , 
Thanks again for the explanations, it worked for me :)
Cheers,
K
